tests/transforms/test_optimization/test_single_qubit_fusion.py

```diff
@@ -0,0 +1,243 @@
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import pytest
+from pennylane import numpy as np
+
+import pennylane as qml
+from pennylane.wires import Wires
+from pennylane.transforms.optimization import single_qubit_fusion
+
+from utils import *
+
+
+class TestSingleQubitFusion:
+    """Test that sequences of any single-qubit rotations are fully fused."""
+
+    def test_single_qubit_full_fusion(self):
+        """Test that a sequence of single-qubit gates all fuse."""
+
+        def qfunc():
+            qml.RZ(0.3, wires=0)
+            qml.Hadamard(wires=0)
+            qml.Rot(0.1, 0.2, 0.3, wires=0)
+            qml.RX(0.1, wires=0)
+            qml.SX(wires=0)
+            qml.T(wires=0)
+            qml.PauliX(wires=0)
+
+        transformed_qfunc = single_qubit_fusion()(qfunc)
+
+        original_ops = qml.transforms.make_tape(qfunc)().operations
+        transformed_ops = qml.transforms.make_tape(transformed_qfunc)().operations
+
+        assert len(transformed_ops) == 1
+
+        # Compare matrices
+        matrix_expected = compute_matrix_from_ops_one_qubit(original_ops)
+        matrix_obtained = compute_matrix_from_ops_one_qubit(transformed_ops)
+        assert check_matrix_equivalence(matrix_expected, matrix_obtained)
```

Comment:
I just realized, we should probably have a function that is

```python
unitary_matrix = qml.get_unitary(qfunc)(params)
```

Reply:
Maybe worth making a good first issue for?

Reply:
Oh good idea!! I'd originally been thinking what is expresesd in with #1313 but it would be way better to have it as a transform instead.